extensions/Lily/Video.js

```diff
@@ -0,0 +1,441 @@
+// Name: Video
+// ID: lmsVideo
+// Description: Play videos from URLs.
+// By: LilyMakesThings <https://scratch.mit.edu/users/LilyMakesThings/>
+
+// Attribution is not required, but greatly appreciated.
+
+(function (Scratch) {
+  "use strict";
+
+  const vm = Scratch.vm;
+  const runtime = vm.runtime;
+  const Cast = Scratch.Cast;
+
+  class Video {
+    constructor() {
+      this.videos = [];
+      this.targets = [];
+
+      runtime.on("PROJECT_STOP_ALL", () => {
+        for (const name of Object.keys(this.videos)) {
+          const video = this.videos[name].video;
+          video.pause();
+          video.currentTime = 0;
+        }
+
+        for (const id of Object.keys(this.targets)) {
+          const target = this.targets[id].target;
+          target.updateAllDrawableProperties();
+        }
+
+        this.targets = [];
+      });
+
+      runtime.on("PROJECT_START", () => {
+        for (const name of Object.keys(this.videos)) {
+          const video = this.videos[name].video;
+          video.pause();
+          video.currentTime = 0;
+        }
+
+        for (const id of Object.keys(this.targets)) {
+          const target = this.targets[id].target;
+          target.updateAllDrawableProperties();
+        }
+
+        this.targets = [];
+      });
+
+      runtime.on("BEFORE_EXECUTE", () => {
+        for (const name of Object.keys(this.videos)) {
+          const videoObject = this.videos[name];
+
+          const video = videoObject.video;
+          const videoVolume = videoObject.volume;
+          const skin = videoObject.skin;
+
+          const projectVolume = runtime.audioEngine.inputNode.gain.value;
+          video.volume = videoVolume * projectVolume;
+
+          vm.renderer.updateBitmapSkin(skin, video, 1);

```

Comment:
Will this be constantly reuploading videos that are paused and not even visible?

Reply:
...probably

Reply:
I added a check for if any drawable has the video applied or if the video is paused. I encourage you to check it yourself in-editor.